probe-rs/src/architecture/arm/sequences.rs

```diff
@@ -633,7 +646,24 @@ pub trait ArmDebugSequence: Send + Sync + Debug {
                 // Setting this on MINDP is unpredictable.
                 ctrl.set_mask_lane(0b1111);
             }
-            interface.write_dp_register(dp, ctrl.clone())?;
+
+            match interface
+                .write_dp_register(dp, ctrl.clone())
+                .and_then(|_| interface.flush())
+            {
+                Ok(()) => {}
+                Err(e @ ArmError::Dap(DapError::NoAcknowledge)) => {
+                    let Some(probe) = interface.try_dap_probe_mut() else {
+                        return Err(e);
```

Comment:
Can we introduce a warn here or a debug? I feel like if this case is ever hit it's hard to follow the error path.